routers/ai.js

```diff
@@ -32,138 +15,272 @@ router.post('/init', async function(req, res) {
+    sceneHierarchy: `The current scene hierarchy includes both 'objects' and their 'child applications' (otherwise known as tools). Objects serve as anchors for the spatial applications that users interact with. Child applications often have a summary of their state, and a list of spatial references within the application that are included in the scene hierarchy. Use this information to understand the placement and configuration of applications relative to their parent objects in the scene, facilitating accurate responses to user queries about spatial interactions within the digital twin environment. Whenever possible, respond with the exact UUIDs of the applications, objects, or spatial references that you are referring to. These UUIDs are keys (satisfying the regular expression ${toolRegex}) in the scene hierarchy and will be used to link your response to additional information. When summarizing the scene hierarchy, you do not need to tell users the exact [X,Y,Z] coordinates of anything, unless specifically asked to.`,
+    interactionLog: `Below is the cumulative log of interactions within the space for the current session, detailing tool additions and removals. Remember, each logged addition indicates the presence of that tool in the space. If the user asks for a summary of the session, or how the system changed over time, the interaction log is a good source of this information. Please include the ID of the user and the ID of the tool in your responses involving the interaction log.`,
+    spatialLogic: `Resolve spatial location queries, like "the position of the spatial cursor of User A", from descriptions to [X,Y,Z] coordinates using a right-handed millimeter coordinate system. Users generally don't understand coordinates presented as numbers, so only respond with coordinates in [X,Y,Z] format if a user specifically asks for the coordinates, or if their request involves function-calling.`,
+    functionGuidelines: `The user may make a request that requires one or more function calls to fulfill it. Function calls are expensive. Before performing a function call, please prepare a plan of the sequence of function calls that you will need to perform to fulfill the request, and ask the user to verify the plan and give you explicit permission to call them. Upon receiving permission, please perform the sequence of function calls sequentially until all of them have been completed, then tell the user the results when all have been completed.`,
+    // functionGuidelines: `Function calls are triggered by explicit user requests or commands implying a series of actions, like "draw a blue square of size 1000". Interpret such commands as complete tasks to be executed in sequence, seeking clarification only if the user's intent or actions are unclear.`,
```

Comment:
Note to anyone testing the system (@Steve-KX-RL , @hobinjk-ptc , @dangond-ptc) - you may want to swap the function guidelines back to the commented version, or to something like:

functionGuidelines: `The user may make a request that requires one or more function calls to fulfill it. Function calls are expensive. First decide if only one function call would be needed to fulfill the request. If so, make the request without asking for permission. But if the request requires multiple function calls, before performing the function calls, please prepare a plan of the sequence of function calls that you will need to perform to fulfill the request, and ask the user to verify the plan and give you explicit permission to call them. Upon receiving permission, please perform the sequence of function calls sequentially until all of them have been completed, then tell the user the results when all have been completed.`

...as the current version requires a confirmation step before every API call, which helps to structure its thoughts about complex sequences of calls, but gets in the way of simpler tasks.